src/CTA.Rules.Common/Extensions/XDocumentExtensions.cs

```diff
@@ -20,5 +22,10 @@ public static bool ContainsAttributeValue(this XDocument document, string elemen
         {
             return document.XPathSelectElements(elementPath).Any(e => e.Attribute(attributeName)?.Value == value);
         }
+
+        public static IEnumerable<XElement> GetElementsByLocalName(this XDocument document, string elementName)
+        {
+            return document.Descendants().Where(d => string.Equals(d.Name.LocalName, elementName, StringComparison.InvariantCulture));
```

Comment:
What if the document doesn't have any Descendants? Empty document? Would this still behave as intended through the stack trace?

Reply:
Good questions. So not having any descendants is the same as have an empty document, in which case this method would yield an empty enumerable which is the expected behavior. In the case of having a single element, `document.Descendants()` would yield an enumerable with just that one element.